libutils/logging.c

```diff
@@ -273,7 +275,34 @@ const char *GetErrorStr(void)
 {
     return strerror(errno);
 }
-#endif
+
+#else
+
+const char *GetErrorStrFromCode(int error_code)
+{
+    static char errbuf[CF_BUFSIZE];
+    int len;
+
+    memset(errbuf, 0, sizeof(errbuf));
+
+    if (FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM, NULL, error_code,
+                  MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US), errbuf, CF_BUFSIZE, NULL))
+    {
+        // remove CRLF from end
+        len = strlen(errbuf);
+        errbuf[len - 2] = errbuf[len - 1] = '\0';
```

Comment:
What if `len` is `0`?

Reply:
No idea, I'm not touching this code :smile:

Reply:
(just moved it around)